go/logic/migrator.go

```diff
@@ -1135,6 +1140,9 @@ func (this *Migrator) iterateChunks() error {
 		}
 		// Enqueue copy operation; to be executed by executeWriteFuncs()
 		this.copyRowsQueue <- copyRowsFunc
+		if !hasFurtherRange {
+			break
+		}
```

Comment:
Since this code is outside `copyRowsFunc`, then it only skips enqueuing of new `copyRowsFunc` functions, but still there could be multiple `copyRowsFunc` instacnes inside `this.copyRowsQueue` that haven't been evaluated yet.

This is why I believe this PR _mitigates_ the potential for deadlock but does not actually _solve it_.

I think we should check `hasFurtherRange` from _within_ `copyRowsFunc` to solve it completely.

Reply:
@shlomi-noach yes, you are right. I will make some modification.